lib/types/zpool.nix

```diff
@@ -13,13 +22,7 @@
       description = "Type";
     };
     mode = lib.mkOption {
-      type = lib.types.enum [
-        ""
-        "mirror"
-        "raidz"
-        "raidz2"
-        "raidz3"
-      ];
+      type = lib.types.enum (modeOptions ++ [ "prescribed" ]);
```

Comment:
If a user provides a `topology`, can we than not default the value to `prescribed`?
I also wonder if we should just call it `topology` here as well for consistency.

Reply:
Different idea. Could we accept the topology as an argument here? I don't know if this works with type checking though.